src/code_800E9A8.c

```diff
@@ -1,11 +1,131 @@
 #include "global.h"
+#include "structs/code_800E9E4.h"
+#include "code_800DAC0.h"
 #include "code_800E9A8.h"
 #include "code_800E9E4.h"
+#include "structs/str_position.h"
 
 s32 sub_800E2C0(u32);
 
 struct unkStruct_203B0CC *gUnknown_203B0CC;
 
+void sub_800DE38(struct unkStruct_203B0CC_sub *);
+void sub_800DE8C(struct unkStruct_203B0CC_sub *, DungeonPos *);
+void sub_800DD0C(struct unkStruct_203B0CC_sub *, DungeonPos *);
+void sub_800DCD0(struct unkStruct_203B0CC_sub *);
+
+s32 sub_800E900(s32 r0);
+s32 sub_800E208(s32, unkStruct_800E208*);
+
+bool8 sub_800E7D0(u16 *param_1)
+{
+    int local_1c [4];
+    bool8 flag;
+    s32 index;
+
+    flag = FALSE;
+    local_1c[0] = sub_800ECB8(*param_1)->unk0;
+    local_1c[1] = sub_800ECB8(*param_1)->unk2;
+    local_1c[2] = sub_800ECB8(*param_1)->unk4;
+    local_1c[3] = sub_800ECB8(*param_1)->unk6;
+    for(index = 0; index < 4; index++)
+    {
+        flag = (sub_800ECA4(local_1c[index])->animType == 4);
+        if(flag) break;
+    }
+    return flag;
+}
+
+bool8 sub_800E838(u16 *param_1, s32 param_2)
+{
+    s32 local_1c [4];
+
+    local_1c[0] = sub_800ECB8(*param_1)->unk0;
+    local_1c[1] = sub_800ECB8(*param_1)->unk2;
+    local_1c[2] = sub_800ECB8(*param_1)->unk4;
+    local_1c[3] = sub_800ECB8(*param_1)->unk6;
+    return sub_800ECA4(local_1c[param_2])->animType == 4;
+}
+
+// TODO: look at this later
```

Comment:
Looks like this is matched. Is there anything else to do for the `TODO`?

Reply:
It's matched but I forced it with the `void *`. I'm not very confident which is why I added the TODO and decomp.me to return to later